ntoskrnl/ke/gate.c

```diff
@@ -197,7 +200,10 @@ KeSignalGateBoostPriority(IN PKGATE Gate)
             /* Release the thread lock */
             KiReleaseThreadLock(WaitThread);
 
-            /* FIXME: Boosting */
+            /* Apply a priority boost */
+            Priority = KiComputeNewPriority(CurrentThread, 0);
```

Comment:
How is the second parameter ("adjustment") chosen?

Reply:
Just to be sure, here you compute the priority based on "CurrentThread", but apply it to another thread (WaitThread)?
(I'm asking because all of our other use-cases of KiComputeNewPriority take a thread in input to calculate the priority, and reapply that new priority on that same thread.)

Reply:
> How is the second parameter ("adjustment") chosen?

This adjustment is an additional parameter, and if it != 0, it is subtracted from the calculated priority, which is not needed in this case (so, CurrentThread->Priority - CurrentThread->PriorityDecrement - 0).

> Just to be sure, here you compute the priority based on "CurrentThread", but apply it to another thread (WaitThread)?

Yes, that way the priority boost is applied to the waiter (or even woken) thread.